This is something I added locally during a really frustrating debug session with some RDL feature. I was about to rebase it out of existence and thought I'd submit it here.  I like that I can easily copy and paste the curl statement (without manually mucking with adding the body) and also refer to the request body.

This is what a failed test looks like if you specify RDC_PP=True:

<pre><code>
======================================================================
FAIL: test_create (tests.api.instances.CreateInstance)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/hgfs/Ridley/reddwarf_lite-integration/tests/integration/tests/api/instances.py", line 297, in test_create
    fail("FAIL!")
AssertionError: FAIL!
-------------------- >> begin captured logging 
reddwarfclient.client: DEBUG: REQUEST:
reddwarfclient.client: DEBUG: curl -i http://localhost:8779/v1.0/6e45a760df154dbab4fb041e65a30d9c/instances -X POST -H "X-Auth-Project-Id: reddwarf" -H "User-Agent: python-reddwarfclient" -H "Content-Type: application/json" -H "Accept: application/json" -H "X-Auth-Token: f8c105a28c4e494487d5828f233a771f" -d '{"instance": {"volume": null, "users": [{"password": "litepass", "name": "lite", "databases": [{"name": "firstdb"}]}], "flavorRef": "https://localhost:8779/v1.0/6e45a760df154dbab4fb041e65a30d9c/flavors/7", "name": "TEST_2012-07-10 09:02:26.720840", "databases": [{"collate": "latin2_general_ci", "name": "firstdb", "character_set": "latin2"}, {"name": "db2"}]}}'
reddwarfclient.client: DEBUG: BODY: {
    "instance": {
        "databases": [
            {
                "character_set": "latin2", 
                "collate": "latin2_general_ci", 
                "name": "firstdb"
            }, 
            {
                "name": "db2"
            }
        ], 
        "flavorRef": "https://localhost:8779/v1.0/6e45a760df154dbab4fb041e65a30d9c/flavors/7", 
        "name": "TEST_2012-07-10 09:02:26.720840", 
        "users": [
            {
                "databases": [
                    {
                        "name": "firstdb"
                    }
                ], 
                "name": "lite", 
                "password": "litepass"
            }
        ], 
        "volume": null
    }
}

reddwarfclient.client: DEBUG: RESPONSE HEADERS: {'date': 'Tue, 10 Jul 2012 16:02:26 GMT', 'status': '200', 'content-length': '663', 'content-type': 'application/json'}
reddwarfclient.client: DEBUG: RESPONSE BODY   : {
    "instance": {
        "created": "2012-07-10T16:02:26", 
        "flavor": {
            "id": "7", 
            "links": [
                {
                    "href": "https://localhost:8779/v1.0/6e45a760df154dbab4fb041e65a30d9c/flavors/7", 
                    "rel": "self"
                }, 
                {
                    "href": "https://localhost:8779/flavors/7", 
                    "rel": "bookmark"
                }
            ]
        }, 
        "id": "0c200579-cda7-4aa6-914c-1dcb04beaf93", 
        "links": [
            {
                "href": "https://localhost:8779/v1.0/6e45a760df154dbab4fb041e65a30d9c/instances/0c200579-cda7-4aa6-914c-1dcb04beaf93", 
                "rel": "self"
            }, 
            {
                "href": "https://localhost:8779/instances/0c200579-cda7-4aa6-914c-1dcb04beaf93", 
                "rel": "bookmark"
            }
        ], 
        "name": "TEST_2012-07-10 09:02:26.720840", 
        "status": "BUILD", 
        "updated": "2012-07-10T16:02:26", 
        "volume": {
            "size": null
        }
    }
}
---------------------  end captured logging << ---------------------

</code></pre>

--- a/reddwarfclient/client.py
+++ b/reddwarfclient/client.py
@@ -34,6 +34,9 @@ from reddwarfclient import exceptions
 
 
 _logger = logging.getLogger(__name__)
+RDC_PP = os.environ.get("RDC_PP", "False") == "True"
+
+
 if 'REDDWARFCLIENT_DEBUG' in os.environ and os.environ['REDDWARFCLIENT_DEBUG']:
     ch = logging.StreamHandler()
     _logger.setLevel(logging.DEBUG)
@@ -86,6 +89,12 @@ class ReddwarfHTTPClient(httplib2.Http):
         return self.times
 
     def http_log(self, args, kwargs, resp, body):
+        if not RDC_PP:
+            self.simple_log(args, kwargs, resp, body)
+        else:
+            self.pretty_log(args, kwargs, resp, body)
+
+    def simple_log(self, args, kwargs, resp, body):
         if not _logger.isEnabledFor(logging.DEBUG):
             return
 
@@ -104,6 +113,42 @@ class ReddwarfHTTPClient(httplib2.Http):
         if 'body' in kwargs:
             _logger.debug("REQ BODY: %s\n" % (kwargs['body']))
         _logger.debug("RESP:%s %s\n", resp, body)
+
+    def pretty_log(self, args, kwargs, resp, body):
+        from reddwarfclient import common
+        if not _logger.isEnabledFor(logging.DEBUG):
+            return
+
+        string_parts = ['curl -i']
+        for element in args:
+            if element in ('GET', 'POST'):
+                string_parts.append(' -X %s' % element)
+            else:
+                string_parts.append(' %s' % element)
+
+        for element in kwargs['headers']:
+            header = ' -H "%s: %s"' % (element, kwargs['headers'][element])
+            string_parts.append(header)
+
+        curl_cmd = "".join(string_parts)
+        _logger.debug("REQUEST:")
+        if 'body' in kwargs:
+            _logger.debug("%s -d '%s'" % (curl_cmd, kwargs['body']))
+            try:
+                req_body = json.dumps(json.loads(kwargs['body']),
+                                      sort_keys=True, indent=4)
+            except:
+                req_body = kwargs['body']
+            _logger.debug("BODY: %s\n" % (req_body))
+        else:
+            _logger.debug(curl_cmd)
+
+        try:
+            resp_body = json.dumps(json.loads(body), sort_keys=True, indent=4)
+        except:
+            resp_body = body
+        _logger.debug("RESPONSE HEADERS: %s" % resp)
+        _logger.debug("RESPONSE BODY   : %s" % resp_body)
 
     def request(self, *args, **kwargs):
         kwargs.setdefault('headers', kwargs.get('headers', {}))

